recipe/gen_patch_json.py

```diff
@@ -344,6 +344,42 @@ def _gen_patch_instructions(index, new_index, subdir):
     return instructions
 
 
+# this function is the old version of _pin_stricter
+# the version used by the patch YAML code is updated to handle
+# additional cases
+def _pin_stricter(fn, record, fix_dep, max_pin, upper_bound=None):
```

Comment:
@jaimergp See here for a version of pin_stricter that handles more cases. No need to use it.